### DISCUSSION TITLE
Allow empty migrations to throw error instead of just `pass`

### DISCUSSION THREAD

**Discussion by jd-solanki** [sqlalchemy/alembic#1550] [Ideas]
Hi 👋🏻 

I switched from Django to FastAPI and glad to know that we have alembic for SQLAlchemy. This is my first discussion/issue in this repo.

I work in a team and most of the people aren't much experienced with alembic so sometimes they accidently adds multiple migration files and sometimes few contains blank migration script and when we run migrate it gets used unknownly.

It will be great if we can strict approach where migration script should atleast do something else raise exception. For example, When we generate the migration and alembic can't detect any changes instead of just `pass`:
```py
def upgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    pass
    # ### end Alembic commands ###


def downgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    pass
    # ### end Alembic commands ###
```

Can we add error something like:
```py
def upgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    raise NotImplementedError
    # ### end Alembic commands ###


def downgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    raise NotImplementedError
    # ### end Alembic commands ###
```

This will mitigate the unwanted empty migration scripts.

I tried to search it in config options but couldn't find it.

**Comment by CaselIT** [comment]
Hi,

You can already do this using the alembic api. There is a similar example in the cookbook:
https://alembic.sqlalchemy.org/en/latest/cookbook.html#cookbook-no-empty-migrations

**Reply by jd-solanki** [reply]
Woah! Sorry for the noise 🙏🏻 

> P.S. It's hard to look for things in SA & Alembic docs. Any plans to move to MkDoc Material like other projects?

**Reply by CaselIT** [reply]
No. Almebic doc were just changed to the current theme, that seems mostly similar to what you are suggesting
